api/destination.go

```diff
@@ -61,6 +64,9 @@ func (r CreateDestinationRequest) ValidationRules() []validate.ValidationRule {
 		validate.Required("uniqueID", r.UniqueID),
 		validateDestinationName(r.Name),
 		validate.Required("name", r.Name),
+		// Allow "" for versions 0.16.1 and prior
+		// TODO: make this required in the future
+		validate.Enum("kind", r.Kind, []string{"kubernetes", "ssh", ""}),
```

Comment:
We may want to keep this optional (and set it to `kubernetes` if unset for backwards compat). To make infra extensible it may be helpful for users to be able to set this to an arbitrary string (maybe with some format limitations)

Reply:
SGTM - well put